src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/name-interaction.mdx

```diff
@@ -0,0 +1,201 @@
+tags:
+  - Mobile monitoring
+  - Mobile SDK API
+  - Custom instrumentation
+metaDescription: Mobile app monitoring API used to sets the name to rename default interaction names reported to New Relic.
+redirects:
+  - /docs/mobile-monitoring/new-relic-mobile-ios/ios-sdk-api/custom-interaction-name
+  - /docs/mobile-monitoring/new-relic-mobile-ios/ios-sdk-api/custom-interaction-name-ios-sdk-api
+  - /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/setinteractionname-android-agent-api
+  - /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/setinteractionname-android-sdk-api
+  - /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/set-interaction-name
+---
+
+<Tabs>
+	<TabsBar>
+        <TabsBarItem id="android">
+            Android
+        </TabsBarItem>
+        <TabsBarItem id="ios">
+            iOS
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java] 
+
+```java
+NewRelic.setInteractionName(string $interactionName)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+  NewRelic.setInteractionName(name: String?)
+```
+
+## Description [#description]
+
+Set a new name for an interaction that is already being tracked by New Relic.
+
+Use `setInteractionName()` to change the name of an interaction in an instrumented app.
+
+For example, you have an interaction that is being reported under a single activity name, like `FragmentActivity`, or under an obfuscated name, like `baseclass.a`, and you want to rename the interaction to be more descriptive,. You could use `setInteractionName` at the beginning of each `onCreate()` method to change the name.
+
+To create a new interaction, see `startInteraction()`.
+
+For general info on using this API, see the [Android SDK API usage guide](/docs/mobile-monitoring/new-relic-mobile-android/install-configure/work-android-sdk-api).
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$interactionName`
+      </td>
+
+            <td>
+        `string`
+      </td>
+
+      <td>
+        Required. The name you want to give to the interaction.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Example [#example]
+
+This example uses `NewRelic.setInteractionName()` at the beginning of the `onCreate()` method as part of the `Activity` class:
+
+### Java [#java]
+
+```java
+public class MainActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+    	//Rename the in-flight interaction
+        NewRelic.setInteractionName("Display MyCustomInteraction");
+    }
+
+    // ... continue methods ...
+
+}
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
NewRelic.setInteractionName("Display MyCustomInteraction")

Reply:
```suggestion
```kotlin

NewRelic.setInteractionName("Display MyCustomInteraction")

```
```